docs/source/extensions/cli.py

```diff
@@ -0,0 +1,127 @@
+
+
+def _align_directive(argument: str) -> str:
+    return directives.choice(argument, ("left", "center", "right"))
+
+
+class CLIDirective(SphinxDirective):
+    """
+    Directive for listing CLI commands in a table.
+    """
+    required_arguments = 1  # The import path of the CLI
+    option_spec = {
+        "align": _align_directive,
+    }
+
+    def run(self) -> list[nodes.Node]:
+        """
+        Process the content of the shield directive.
+        """
+        align: Literal["left", "center", "right"] | None = self.options.get("align")
+
+        headers = [nodes.entry("", nodes.paragraph("", header)) for header in ["Command", "Description"]]
+
+        import_string, = self.arguments
+        cli = self.import_cli(import_string)
+
+        rows: list[list[nodes.entry]] = []
+
+        for command_name, command in cli.commands.items():
+            rows.append([
+                nodes.entry("", nodes.paragraph("", "", nodes.reference("", "", nodes.literal("", command_name), refid=f"cli.{command_name}"))),
+                nodes.entry("", nodes.paragraph("", command.description or "")),
+            ])
+
+        table = self.create_table(headers, rows, align=align)
+
+        nodes_to_return: list[nodes.Node] = [table]
+
+        for command_name, command in cli.commands.items():
+            command_section = nodes.section(ids=[f"cli.{command_name}"])
+            title = nodes.title(f"cli.{command_name}", "", nodes.literal("", command_name))
+            command_section.append(title)
+
+            if command.description is not None:
+                command_section.append(nodes.paragraph("", command.description))
+
+            usage_block = nodes.literal_block("", f"$ {command.usage_string()}")
+            command_section.append(usage_block)
+
+            for positional in command.positional_arguments:
+                alias_paragraph = nodes.paragraph("", "", nodes.literal("", positional.name))
+                description_paragraph = nodes.paragraph("", positional.description or "")
+                description_paragraph.set_class("cli-option-description")
+                command_section.extend([alias_paragraph, description_paragraph])
+
+            for option in command.options:
+                alias_nodes: list[nodes.Node] = []
+                for alias in option.aliases:
+                    alias_nodes.append(nodes.literal("", alias))
+                    alias_nodes.append(nodes.Text(", "))
+                alias_paragraph = nodes.paragraph("", "", *alias_nodes[:-1])
+                description_paragraph = nodes.paragraph("", option.description or "")
+                description_paragraph.set_class("cli-option-description")
+                command_section.extend([alias_paragraph, description_paragraph])
+
+            nodes_to_return.append(command_section)
+
+        return nodes_to_return
+
+    def create_table(self, headers: list[nodes.entry], rows: list[list[nodes.entry]],
```

Comment:
Do we need some Python Docs here?

Reply:
Since this is a documentation function as opposed to a library function, I would suggest that it's sufficiently self documenting based on the function name and parameter names.  Happy to be overruled on that though?